### PR TITLE
[CARBONDATA-429] Remove unnecessary file name check in dictionary cache

### PR DESCRIPTION
**Pull request by jackylk**
In dictionary cache, there are currently unnecessary file name check for each column, which increase the number of HDFS interactions unnecessarily.
In this PR, file name check is replaced by primitive type check, and returns null if it is a complex type.

One test case (`testGetForwardDictionaryCacheWithException`) is removed, it is expecting an exception when getting the dictionary from cache when the cache is not there. It is invalid test case now as the file name check is removed

### CODE REVIEW

**Comment by ravipesala** [comment]
Build finished.

File: core/src/main/java/org/apache/carbondata/core/cache/dictionary/ReverseDictionaryCache.java [hunk 1]
```diff
@@ -167,12 +167,9 @@ private Dictionary getDictionary(
...
     Dictionary reverseDictionary = null;
-    // create column dictionary info object only if dictionary and its
-    // metadata file exists for a given column identifier
-    if (!isFileExistsForGivenColumn(dictionaryColumnUniqueIdentifier)) {
-      throw new CarbonUtilException(
-          "Either dictionary or its metadata does not exist for column identifier :: "
-              + dictionaryColumnUniqueIdentifier.getColumnIdentifier());
+    // create column dictionary info object only if it is primitive type.
+    if (dictionaryColumnUniqueIdentifier.getDataType().isComplexType()) {
+      return null;
```
Comment (piaoyats) [inline core/src/main/java/org/apache/carbondata/core/cache/dictionary/ReverseDictionaryCache.java]: I am not sure this condition check is right or not? Maybe we still need to do dictionary encoding for complex type?
@QiangCai  pls review this logic

Comment (QiangCai) [inline core/src/main/java/org/apache/carbondata/core/cache/dictionary/ReverseDictionaryCache.java]: We will not invoke getDictionary() method on  complex type column directly.  For complex type column, we call getDictionary() method on primitive type sub-column.

Comment (piaoyats) [inline core/src/main/java/org/apache/carbondata/core/cache/dictionary/ReverseDictionaryCache.java]: So, here logic is right.

**Comment by piaoyats** [comment]
LGTM

**Comment by chenliang613** [comment]
test this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.5.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/191/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.5.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/230/

File: core/src/main/java/org/apache/carbondata/core/cache/dictionary/ReverseDictionaryCache.java [hunk 2]
```diff
@@ -167,12 +167,9 @@ private Dictionary getDictionary(
...
       throws CarbonUtilException {
     Dictionary reverseDictionary = null;
-    // create column dictionary info object only if dictionary and its
-    // metadata file exists for a given column identifier
-    if (!isFileExistsForGivenColumn(dictionaryColumnUniqueIdentifier)) {
-      throw new CarbonUtilException(
-          "Either dictionary or its metadata does not exist for column identifier :: "
-              + dictionaryColumnUniqueIdentifier.getColumnIdentifier());
+    // create column dictionary info object only if it is primitive type.
+    if (dictionaryColumnUniqueIdentifier.getDataType().isComplexType()) {
```
Comment (kumarvishal09) [inline core/src/main/java/org/apache/carbondata/core/cache/dictionary/ReverseDictionaryCache.java]: Do we need this check?? for complex data type column like array and struct no need to invoke this class, as dictionary will be generated only for primitive type column.

Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/cache/dictionary/ReverseDictionaryCache.java]: ok, I will change it to use assert

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.5.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/236/

**Pull request closed without merging**